src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -210,6 +215,30 @@ public static IResourceBuilder<AzureStorageEmulatorResource> WithTablePort(this
         });
     }
 
+    /// <summary>
+    /// Ensures the emulator checks that the requested API version is valid.
+    /// </summary>
+    /// <param name="builder">Storage emulator resource builder.</param>
+    /// <param name="enable">Whether to enable API version check or not. Default is <lang>true</lang>.</param>
+    /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureStorageEmulatorResource"/>.</returns>
+    public static IResourceBuilder<AzureStorageEmulatorResource> WithApiVersionCheck(this IResourceBuilder<AzureStorageEmulatorResource> builder, bool enable = true)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder
+            .WithArgs(context =>
+        {
+            context.Args.Remove(SkipApiVersionCheckArgument);
+
+            if (enable)
+            {
+                context.Args.Add(SkipApiVersionCheckArgument);
+            }
+
+            return Task.CompletedTask;
+        });
```

Comment:
Should this be driven off an annotation instead? This is kind of a hack to remove it, and then add it back.